src/tactics.ml

```diff
@@ -103,6 +103,42 @@ let handle_tactic : sig_state -> Proof.t -> p_tactic -> Proof.t =
       handle_refine (Rewrite.reflexivity tac.pos ps)
   | P_tac_sym           ->
       handle_refine (Rewrite.symmetry tac.pos ps)
+  | P_tac_why3(s)    ->
+      (* get the goal to prove *)
+      let (hypothesis, trm) = Proof.Goal.get_type g in
+      (* get the default or the indicated name of the prover. *)
+      let prover_name =
+        match s with
+        | None          -> Timed.(!Why3prover.current_prover)
+        | Some(name)    -> name
+      in
+      (* translate from lambdapi to why3 terms. *)
+      let (constants_table, hyps, why3term) =
+          Why3prop.translate tac.pos ps.proof_builtins (hypothesis, trm) in
+      (* create a new task that contains symbols, axioms and the goal. *)
+      let tsk = Why3task.create constants_table hyps why3term in
+      (* call the prover named [prover_name] and get the result. *)
+      let prover_result = Why3prover.call tac.pos prover_name tsk in
+      (* if the prover succeed to prover the goal. *)
```

Comment:
to prove

Reply:
Done in a29414085332857be96b62ab67a96960b46d746d.